daemon/protocol/client_lib.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python
+
+import os
+import re
+import tempfile
+import shutil
+
+from grpc.beta import implementations
+#import grpc
+import daemon_pb2
+
+
+# Copied from daemon/protocol/locate.go
+def _locate_scoot_dir():
+    scootdir = os.environ.get("SCOOTDIR")
+    if scootdir is not None:
+        return scootdir
+    homedir = os.environ.get("HOME")
+    if homedir is not None:
+        return os.path.join(homedir, ".scoot")
+    return None
+
+def _locate_socket():
+    scootdir = _locate_scoot_dir()
+    if scootdir is None:
+        return None
+    return os.path.join(scootdir, "socket")
+
+
+# Singleton variables.
+_domain_sock = _locate_socket()
+_client = None
+_context = None
+
+class ScootException(Exception):
+    """
+    """
+    def __init__(self, caught):
+        self.caught = caught
+
+
+
+class ScootStatus:
+    """
+    """
+    class State:
+        UNKNOWN = 0
+        PENDING = 1
+        PREPARING = 2
+        RUNNING = 3
+        COMPLETED = 4
+        FAILED = 5
+    def __init__(self, run_id, state, snapshot_id, exit_code, error):
+        self.run_id = run_id
+        self.state = state
+        self.snapshot_id = snapshot_id
+        self.exit_code = exit_code
+        self.error = error
+
+
+def start():
+    """
+    """
+    global _client, _domain_sock
+    if _domain_sock is None:
+        raise ScootExceptoin("Could not find domain socket path.")
```

Comment:
here and elsewhere, should this actually be:

`raise ScootException(Exception("Could not find domain socket path"))`

given that `ScootException` expects to wrap an exception instance vs a string?